backend/apps/github/common.py

```diff
@@ -227,3 +232,61 @@ def sync_repository(
     )
 
     return organization, repository
+
+
+def sync_issue_comments(gh_client: Github, issue: Issue):
+    """Sync new comments for a mentorship program specific issue on-demand.
+
+    Args:
+        gh_client (Github): GitHub client.
+        issue (Issue): The local database Issue object to sync comments for.
+
+    """
+    logger.info("Starting comment sync for issue #%s", issue.number)
+
+    try:
+        if not (repository := issue.repository):
+            logger.warning("Issue #%s has no repository, skipping", issue.number)
+            return
+
+        logger.info("Fetching repository: %s", repository.path)
+
+        gh_repository = gh_client.get_repo(repository.path)
+        gh_issue = gh_repository.get_issue(number=issue.number)
+
+        since = issue.comments.order_by("-updated_at").values_list(
+            "updated_at", flat=True
+        ).first() or getattr(issue, "updated_at", None)
+
+        comments = []
+
+        gh_comments = gh_issue.get_comments(since=since) if since else gh_issue.get_comments()
+
+        for gh_comment in gh_comments:
+            author = User.update_data(gh_comment.user)
+            if not author:
+                logger.warning("Could not sync author for comment %s", gh_comment.id)
+                continue
+
+            comment = Comment.update_data(gh_comment, author=author, save=False)
+            comment.content_object = issue
```

Comment:
Why it's not a part of Comment::update_data. What's a good reason for setting it here outside of the function that takes care of the data update?